src/slice.rs

```diff
@@ -623,3 +623,273 @@ macro_rules! s(
         &*&$crate::s![@parse ::std::marker::PhantomData::<$crate::Ix0>, [] $($t)*]
     };
 );
+
+/// Returns a ZST representing the lifetime of the mutable view.
+#[doc(hidden)]
+pub fn life_of_view_mut<'a, A, D: Dimension>(
+    _view: &ArrayViewMut<'a, A, D>
+) -> PhantomData<&'a mut A> {
+    PhantomData
+}
+
+/// Derefs the raw mutable view into a view, using the given lifetime.
+#[doc(hidden)]
+pub unsafe fn deref_raw_view_mut_into_view_with_life<'a, A, D: Dimension>(
+    raw: RawArrayViewMut<A, D>,
+    _life: PhantomData<&'a mut A>,
+) -> ArrayView<'a, A, D> {
+    raw.deref_into_view()
+}
+
+/// Derefs the raw mutable view into a mutable view, using the given lifetime.
+#[doc(hidden)]
+pub unsafe fn deref_raw_view_mut_into_view_mut_with_life<'a, A, D: Dimension>(
+    raw: RawArrayViewMut<A, D>,
+    _life: PhantomData<&'a mut A>,
+) -> ArrayViewMut<'a, A, D> {
+    raw.deref_into_view_mut()
+}
+
+/// Take multiple slices simultaneously.
+///
+/// This macro makes it possible to take multiple slices of the same array, as
+/// long as Rust's aliasing rules are followed for *elements* in the slices.
+/// For example, it's possible to take two disjoint, mutable slices of an
+/// array, with one referencing the even-index elements and the other
+/// referencing the odd-index elements. If you tried to achieve this by calling
+/// `.slice_mut()` twice, the borrow checker would complain about mutably
+/// borrowing the array twice (even though it's safe as long as the slices are
+/// disjoint).
+///
+/// The syntax is `multislice!(` *expression, (pattern [, pattern [, …]])* `)`,
+/// where *expression* evaluates to an `ArrayBase<S, D>` where `S: DataMut`,
```

Comment:
I guess you disagree (since you wrote it this way), but I'd prefer to say "mutable array" rather than "`ArrayBase<S, D>` where `S: DataMut`". (Yes I'm constantly unhappy with exposing ArrayBase, DataMut like this :slightly_smiling_face: .)

Reply:
I like "mutable array" better too. I think I originally wrote it that way because macros don't have type signatures like methods do, which tends to make me want to be fairly explicit about the types they expect and return. In this case, though, "mutable array" is clear and less verbose.